lib/plugins/inspec-sign/lib/inspec-sign/base.rb

```diff
@@ -0,0 +1,119 @@
+require "base64" unless defined?(Base64)
+require "openssl" unless defined?(OpenSSL)
+require "pathname" unless defined?(Pathname)
+require "set" unless defined?(Set)
+require "tempfile" unless defined?(Tempfile)
+require "yaml"
+require "inspec/dist"
+require "inspec/utils/json_profile_summary"
+require "inspec/iaf_file"
+
+module InspecPlugins
+  module Sign
+    class Base
+      include Inspec::Dist
+
+      KEY_BITS = 2048
+      KEY_ALG = OpenSSL::PKey::RSA
+
+      INSPEC_PROFILE_VERSION_1 = "INSPEC-PROFILE-1".freeze
+      INSPEC_REPORT_VERSION_1 = "INSPEC-REPORT-1".freeze
+
+      INSPEC_PROFILE_VERSION_2 = "INSPEC-PROFILE-2".freeze
+      ARTIFACT_DIGEST = OpenSSL::Digest::SHA512
+      ARTIFACT_DIGEST_NAME = "SHA512".freeze
+
+      VALID_PROFILE_VERSIONS = Set.new [INSPEC_PROFILE_VERSION_1, INSPEC_PROFILE_VERSION_2]
+      VALID_PROFILE_DIGESTS = Set.new [ARTIFACT_DIGEST_NAME]
+
+      SIGNED_PROFILE_SUFFIX = "iaf".freeze
+      SIGNED_REPORT_SUFFIX = "iar".freeze
+
+      def self.keygen(options)
+        key = KEY_ALG.new KEY_BITS
+        puts "Generating private key"
+        open "#{options["keyname"]}.pem.key", "w" do |io|
+          io.write key.to_pem
+        end
+        puts "Generating public key"
+        open "#{options["keyname"]}.pem.pub", "w" do |io|
+          io.write key.public_key.to_pem
+        end
+      end
+
+      def self.profile_sign(options)
+        artifact = new
+        path_to_profile = options["profile"]
+
+        Dir.mktmpdir do |workdir|
```

Comment:
I don't think we can use a tempdir. This breaks when archive vendors a profile with local path dependencies.